### DISCUSSION TITLE
ValueError: x and y must be the same size

### DISCUSSION THREAD

**Discussion by iesteaa** [mrdbourke/pytorch-deep-learning#885] [Q&A] [answered]
Hello everyone 
I am in the process of learning Pytorch, I got a problem that I can't solve yet. It seems that the error occurs because there is an error in creating a dataset or data visualization, some sources say this error usually occurs because there are different dimensions, but after I checked, all data has the same dimensions. 

Here is the code: 
### dataset prep
    weight = 0.25
    bias = 0.75
    start = 0
    steps = 1
    end = 100
    
    X = torch.arange(start,end,steps).unsqueeze(dim=1)
    X = X.type(torch.float32)
    y = weight*X + bias

### Split Data
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)
    len(X_test), len(X_train), len(y_train), len(y_test)

### Visualize Data
    def plot_prediction(train_data=X_train,
                         train_labels=y_train,
                         test_data=X_test,
                         test_labels=y_test,
                         predictions=None):

    # Plot training,test and prediction data
    plt.figure(figsize=(10, 7))
    
    # Plot training data in blue
    plt.scatter(train_data, train_labels, c="b", s= 10, label="Training data")

    # Plot test data in green
    plt.scatter(test_data, test_labels, c="g", s=10, label="Testing data")

    if predictions is not None:
        # Plot the predictions in red (predictions were made on the test data)
        plt.scatter(test_data, predictions, c="r", s=10, label="Predictions")

    # Show the legend
    plt.legend(prop={"size": 14});  

I've done with the training and testing process, and gonna visualize it . Yap the error occurred : 

![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/89826086/5b77d338-37a0-49cf-bb27-7aea64ee81cc)

Your help will be invaluable

**Comment by garvpatidar04** [comment, marked as answer]
I think the error might be in line 1 `plot_prediction(predictions=test_pred.cpu)`  instead it should be 
 `plot_prediction(predictions=test_pred.cpu())`  because `.cpu()` is a method not an attribute, so you should always put ` () ` at the end of a
 method. 
 I hope this solves the error.

**Reply by iesteaa** [reply]
Than you for your answer, appreciate it. Solved!

**Reply by teidenzero** [reply]
I had the same problem and this solved it, many thanks!

**Comment by LuluW8071** [comment]
After training the code perform below snippet it should give you the plots

```python
with torch.inference_mode():
  test_pred = model(X_test)

# Plot new predicted data
plot_predict(predictions = test_pred)
```
